File: escutador.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void callback_funcao(const std_msgs::String& msg){

        ROS_INFO("Mensagem: %s",msg.data.c_str());

}

int main(int argc, char **argv){
        ros::init(argc,argv,"escucomunicacao");

        ros::NodeHandle nh;

        ros::Subscriber sub = nh.subscribe("/comunicacao", 1000, callback_funcao);

        ros::spin();

}

File: escutador.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String

def callback_funcao(msg):
        rospy.loginfo("Mensagem: ")
        rospy.loginfo(msg)


if __name__ == '__main__':
        rospy.init_node('escucomunicacao')
        sub = rospy.Subscriber('comunicacao', String, callback_funcao)

        rospy.spin()



~                                                                               
~                                                                               
~                                                                               
~                                                                               
~           
